src/CoreWCF.NetTcp/tests/AuthorizationTest.cs

```diff
@@ -0,0 +1,96 @@
+
+        public AuthorizationTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact(Skip ="Skipped in pipeline run")]
+        public void AuthorizationBasedonRolesTest()
+        {
+            string testString = "a" + PrincipalPermissionMode.Always.ToString() + "test";
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<StartupForAuthorization>(_output).Build();
+            using (host)
+            {
+                System.ServiceModel.ChannelFactory<ClientContract.ITestService> factory = null;
+                ClientContract.ITestService channel = null;
+                host.Start();
+                try
+                {
+                    System.ServiceModel.NetTcpBinding binding = ClientHelper.GetBufferedModeBinding(System.ServiceModel.SecurityMode.Transport);
+                    factory = new System.ServiceModel.ChannelFactory<ClientContract.ITestService>(binding,
+                        new System.ServiceModel.EndpointAddress(host.GetNetTcpAddressInUse() + Startup.WindowsAuthRelativePath));
+                    channel = factory.CreateChannel();
+                    ((IChannel)channel).Open();
+                    string result = channel.EchoForAuthorizarionOneRole(testString);
+                    Assert.Equal(testString, result);
+                    try
+                    {
+                        channel.EchoForAuthorizarionNoRole(testString);
+                    }catch(Exception ex)
+                    {
+                        Assert.Contains("Access is denied", ex.Message);
+                    }
+                    ((IChannel)channel).Close();
+                    factory.Close();
+                }
+                finally
+                {
+                    ServiceHelper.CloseServiceModelObjects((IChannel)channel, factory);
+                }
+            }
+        }
+    }
+
+    public class StartupForAuthorization
+    {
+        public const string WindowsAuthRelativePath = "/nettcp.svc/windows-auth";
+        public const string NoSecurityRelativePath = "/nettcp.svc/security-none";
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddServiceModelServices();
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            app.UseServiceModel(builder =>
+            {
+                builder.AddService<Services.TestService>();
+                builder.AddServiceEndpoint<Services.TestService, ServiceContract.ITestService>(new CoreWCF.NetTcpBinding(), WindowsAuthRelativePath);
+                builder.AddServiceEndpoint<Services.TestService, ServiceContract.ITestService>(new CoreWCF.NetTcpBinding(SecurityMode.None), NoSecurityRelativePath);
+                Action<ServiceHostBase> serviceHost = host => ChangeHostBehavior(host);
+                builder.ConfigureServiceHostBase<Services.TestService>(serviceHost);
```

Comment:
You don't need an intermediate variable. You should just be able to pass the method directly to ConfigureServiceHostBase.